src/Auth/AuthProvider.js

```diff
@@ -0,0 +1,169 @@
+import { createContext, useContext } from "react";
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router";
+import axios from "axios";
+
+export const AuthContext = createContext();
+
+export const setupAuthHeaderForServiceCalls = (token) => {
+  if (token) {
+    return (axios.defaults.headers.common["Authorization"] = token);
+  }
+  delete axios.defaults.headers.common["Authorization"];
+};
+
+export const setupAuthExceptionHandler = (logoutUser, navigate) => {
+  const UNAUTHORIZED = 401;
+  axios.interceptors.response.use(
+    (response) => response,
+    (error) => {
+      if (error?.response?.status === UNAUTHORIZED) {
+        logoutUser();
+        console.log("here");
+        navigate("login");
+      }
+      return Promise.reject(error);
+    }
+  );
+};
```

Comment:
you can extract these two functions in separate files in utils

Reply:
Done